frontend/src/pages/RoadTripPage/MapPage/MapModal/MapModal.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useRef, useCallback } from "react";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import ReactMapGL from "react-map-gl";
+import Geocoder from "react-map-gl-geocoder";
+import axios from "axios";
+import { useAuth0 } from "@auth0/auth0-react";
+import "react-map-gl-geocoder/dist/mapbox-gl-geocoder.css";
+
+const dotenv = require("dotenv");
+dotenv.config();
+
+export default function MapModal({
+    open,
+    handleClose,
+    setDestination,
+    setDestSelected,
+}) {
+    // Default location is just New Zealand. If/when user selects a different location this will be overridden
+    const initialViewport = {
+        width: "100vw",
+        height: "100vh",
+        latitude: -41.01633,
+        longitude: 170.83285,
+        zoom: 4,
+    };
+
+    const [viewport, setViewport] = useState(initialViewport);
+    const [destName, setDestName] = useState("New Zealand");
+    const [destLatitude, setDestLatitude] = useState(-41.01633);
+    const [destLongitude, setDestLongitude] = useState(170.83285);
+
+    const mapAccessToken = process.env.REACT_APP_MAPBOX_TOKEN;
+    const { getAccessTokenSilently } = useAuth0();
+    const mapRef = useRef();
+    const geocoderContainerRef = useRef();
+
+    // Size of map inside the modal - unable to split into separate CSS file
+    const mapStyle = {
+        width: "100%",
+        height: 400,
+    };
+
+    const handleViewportChange = useCallback(
+        (newViewport) => setViewport(newViewport),
+        []
+    );
+
+    // 'Travel' to selected location
+    const handleGeocoderViewportChange = useCallback(
+        (newViewport) => {
+            const geocoderDefaultOverrides = { transitionDuration: 1000 };
+
+            return handleViewportChange({
+                ...newViewport,
+                ...geocoderDefaultOverrides,
+            });
+        },
+        [handleViewportChange]
+    );
+
+    // When location selected, save that location's information
+    const getSelectedResult = useCallback(
+        (result) => {
+            setDestName(result.result.place_name);
+            setDestLongitude(result.result.center[0]);
+            setDestLatitude(result.result.center[1]);
+        },
+        [setDestLatitude, setDestLongitude, setDestName]
+    );
+
+    async function handleSubmit() {
+        // POST request to set the destination
+        const accessToken = await getAccessTokenSilently();
```

Comment:
it would be nice to have this call in a wrapper function - similar to usePost I guess but written so it is not a hook. Just an exported async function. For now keep what you have, but it would good when whatever future PR adds that wrapper you change this to use that.

Reply:
Yeah I agree, this would be good to have in future :)